core/src/main/java/hudson/search/Search.java

```diff
@@ -76,7 +76,7 @@ public void doIndex(StaplerRequest req, StaplerResponse rsp) throws IOException,
                     SuggestedItem target = find(index, query, smo);
                     if(target!=null) {
                         // found
-                        rsp.sendRedirect2(a.getUrl()+target.getUrl());
+                        rsp.sendRedirect2(req.getContextPath()+target.getUrl());
```

Comment:
@jglick Indeed `JenkinsRule ` allows you to create folders, but I don't see the way you can move with `JenkinsRule` to the context of a folder to look for a job which is in the dashboard.